### ISSUE TITLE
Integer Overflow GAUGE32, GAUGE in base_client.rb

### ISSUE THREAD

**Issue by pprugger** [logstash-plugins/logstash-input-snmp#64]
Hello!

In the SNMP specification GAUGE and GAUGE32 are defined as 32bit unsigned integers.
In this plugin however GAUGE AND GAUGE32 are casted to integer, which leads to an overflow for values bigger than MAX_INT.

Since the stack doesn't support unsigned datatypes i suggest to cast the types instead to long.

**Changes:**
In lib/logstash/inputs/snmp/base_client.rb

```
when BER::TIMETICKS, BER::COUNTER, BER::COUNTER32, BER::COUNTER64
          variable.toLong
when BER::INTEGER, BER::INTEGER32, BER::GAUGE, BER::GAUGE32
          variable.toInt
```

**Change to:**

```
when BER::TIMETICKS, BER::COUNTER, BER::COUNTER32, BER::COUNTER64, BER::GAUGE, BER::GAUGE32
          variable.toLong
when BER::INTEGER, BER::INTEGER32
          variable.toInt

```

This patch was tested with rspec and on a logstash instance.
Im running this patch on our cluster and it works fine.

Best regards

**Comment by colinsurprenant** [comment]
Thanks @pprugger! good catch! Did you want to submit a PR for this?

**Comment by pprugger** [comment]
Hey Colin!

I will send a PR tomorrow morning.

**Comment by pprugger** [comment]
Good morning.

I sent a pull request but it seems the buildserver has some problems.

**Comment by colinsurprenant** [comment]
Fixed by #65 - Thanks a lot for your help @pprugger

**Comment by colinsurprenant** [comment]
v1.2.1 published.
